//
// This file is part of the WoW++ project.
// 
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 2 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU General Public License for more details.
// 
// You should have received a copy of the GNU General Public License
// along with this program; if not, write to the Free Software 
// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
//
// World of Warcraft, and all World of Warcraft or Warcraft art, images,
// and lore are copyrighted by Blizzard Entertainment, Inc.
// 

syntax = "proto2";
package wowpp.proto;

message TrainerSpellEntry {
	required uint32 spell = 1;
	required uint32 spellcost = 2;
	optional uint32 reqskill = 3;
	optional uint32 reqskillval = 4;
	optional uint32 reqlevel = 5 [default = 1];
}

message TrainerEntry {
	required uint32 id = 1;
	enum TrainerType {
		CLASS_TRAINER = 0;
		MOUNT_TRAINER = 1;
		SKILL_TRAINER = 2;
		PET_TRAINER = 3;
	};
	optional TrainerType type = 2 [default = CLASS_TRAINER];
	optional uint32 classid = 3;
	optional string title = 4;
	repeated TrainerSpellEntry spells = 5;
}

message Trainers {
	repeated TrainerEntry entry = 1;
}
